### PR TITLE
Multiifo config dev mk2

### PR DESCRIPTION
**Pull request by tapaimarton**
These are the config file changes required for https://github.com/gwastro/pycbc/pull/2383 to work.

### CODE REVIEW

**Comment by tdent** [comment]
@tapaimarton it will quickly become confusing if I am making comments on two requests at two different repos at once. please tell me how to proceed

**Comment by tapaimarton** [comment]
I think we should move the discussion here, so @ahnitz can comment on it as well. I can apply the changes to the other repo once we are all ok with it here.

File: O3exp/pipeline/analysis.ini [hunk 1]
```diff
@@ -71,8 +71,27 @@ do-trigger-fitting =
...
 
-[workflow-coincidence-full]
-parallelization-factor = 20
+[workflow-coincidence-full3d]
+; calculates 3-ifo coincidences in triple coincident time
+parallelization-factor = 1
+
+[workflow-coincidence-full2d0]
+; calculates H-L coincidences over all times when H1 and L1 are observing
+parallelization-factor = 1
```
Comment (tdent) [inline O3exp/pipeline/analysis.ini]: btw, why is parallelization-factor changed from 20 to 1 here, is the new code much more efficient or does it not yet handle parallelization?

File: O3exp/pipeline/analysis.ini [hunk 2]
```diff
@@ -236,27 +255,45 @@ smoothing-width = 0.2
...
+ranking-statistic = newsnr
 
-; Parameters to reproduce this file available at
-; https://git.ligo.org/ligo-cbc/pycbc-software/blob/c2fc41b3c32dbaa2fe859b0d5929b050c01c2b91/statistic-files/v1/make_h1l1_phase_time_amp_v1.sh
-statistic-files = file://localhost/cvmfs/oasis.opensciencegrid.org/ligo/sw/pycbc/statistic-files/v1/H1L1-PHASE_TIME_AMP_v1.hdf
+[multiifo_coinc-full3d]
+; time-slide interval in seconds
+timeslide-interval = 0.1
+; reduction factor to make lightning bolts for IFAR plots
+decimation-factor = 5000
```
Comment (tdent) [inline O3exp/pipeline/analysis.ini]: do we need a high decimation factor and/or loudest keep value for triples, in fact?

File: O3exp/pipeline/analysis.ini (hunk 1 above)
Comment (tapaimarton) [reply O3exp/pipeline/analysis.ini]: It handles it, I just wanted to create 1 file for each case while testing. I can set that back to 20.

File: O3exp/pipeline/analysis.ini (hunk 2 above)
Comment (tapaimarton) [reply O3exp/pipeline/analysis.ini]: Not necessarily, I kept everything at the O2 values for these, my aim was to get the code to work first.

File: O3exp/pipeline/plotting.ini [hunk 3]
```diff
@@ -427,7 +427,7 @@ bin-spacing = log
 psd-model = aLIGOZeroDetHighPower
 
 [page_ifar]
-decimation-factor = ${coinc-full|decimation-factor}
+decimation-factor = ${multiifo_coinc-full3d|decimation-factor}
```
Comment (tdent) [inline O3exp/pipeline/plotting.ini]: note that we might not necessarily have the same decimation factor for all coinc jobs.  
(& I wonder do we really need to give the ifar plot code the decimation factor as an option .. ie prefer it to be stored in the coinc trigger files?)

Comment (tapaimarton) [reply O3exp/pipeline/plotting.ini]: I think the decimation factor is stored in the coinc trigger files, (was stored in the old version as well)

File: O3exp/pipeline/analysis.ini (hunk 1 above)
Comment (tdent) [reply O3exp/pipeline/analysis.ini]: ok, but I think we should keep 20 as a standard value unless @ahnitz disagrees or there is a definite reason to change.

File: O3exp/pipeline/analysis.ini [hunk 4]
```diff
@@ -72,6 +72,7 @@ do-trigger-fitting =
 ;background-bins = all:total:lt600
 
 [workflow-coincidence-full]
+timeslide-precedence = H1, L1, V1, K1, I1
```
Comment (tdent) [inline O3exp/pipeline/analysis.ini:76]: can you propose a comment line here to describe what this option does?

Comment (tapaimarton) [reply O3exp/pipeline/analysis.ini:76]: It took more than a line, but I think this description should be clear enough, is it ok?

**Comment by spxiwh** [comment]
Using these config files I see:

```
ConfigParser.NoSectionError: No section: 'multiifo_coinc-full3d'
```

**Comment by spxiwh** [comment]
Is the problem at line 259 of analysis.ini. Think that should be:
```
[coinc-fullinj&coinc-injfull]
; perform little-dog analysis for software injections
timeslide-interval = ${multiifo_coinc-full|timeslide-interval}
```

**Comment by spxiwh** [comment]
With this change a workflow generates ... Seeing what happens now when I run it!

**Comment by tapaimarton** [comment]
Ah, it seems, I missed that while copying the changes into the files to upload

File: O3exp/pipeline/analysis.ini (hunk 4 above)
Comment (tdent) [reply O3exp/pipeline/analysis.ini:76]: @ahnitz @tapaimarton can I check what this actually means, without the 'pivot'/'fixed' jargon : out of any given combination of detectors, with the current implementation *only one* ifo is slid relative to all the others, and that ifo is called 'pivot' ? is that correct?

Comment (ahnitz) [reply O3exp/pipeline/analysis.ini:76]: @tdent Yes, you can think of it that way.

Comment (tdent) [reply O3exp/pipeline/analysis.ini:76]: OK, would there be any objection to documenting it in the ini file simply as this : 

"for a given ifo combination, time shifted coincidences are generated by shifting one ifo relative to all the rest: that ifo is chosen to be the one appearing first in this list"

The user doesn't need to know everything that goes on under the hood given that only the choice of 'pivot' makes a difference at present.  If/when the 'pivot' and 'fixed' designations mean something more specific we can update this.

**Review by tdent — APPROVED** [review]
OK from my side

**Pull request merged**